lib/compat/wordpress-6.0/blocks.php

```diff
@@ -269,3 +269,30 @@ function ( $response ) {
 	);
 }
 add_action( 'rest_api_init', 'gutenberg_rest_comment_set_children_as_embeddable' );
+
+/**
+ * Registers the lock block attribute for block types.
+ *
+ * Once 6.0 is the minimum supported WordPress version for the Gutenberg
+ * plugin, this shim can be removed
+ *
+ * Doesn't need to be backported into Core.
+ *
+ * @param array $args Array of arguments for registering a block type.
+ * @return array $args
+ */
+function gutenberg_register_lock_attribute( $args ) {
+	// Setup attributes if needed.
+	if ( empty( $args['attributes'] ) ) {
```

Comment:
Not a big deal here, but `$arg = array( 'attributes' => array() );` will get overriden here because `empty` will evaluate to `true`.

Reply:
Good call. I will update the check to use `is_array` 👍

Reply:
Updated to use `isset()`. This way, we can avoid PHP notices if the `attributes` key isn't provided when the filter runs.